docs/template_extension/CMakeLists.txt

```diff
@@ -12,9 +12,15 @@ file(GLOB_RECURSE SRC *.cpp)
 
 add_library(${TARGET_NAME} SHARED ${SRC})
 
+target_compile_options(${TARGET_NAME} PRIVATE -std=c++11)
```

Comment:
why do we need this flag? Our cmake already sets this flag in `os_flags.cmake`

Reply:
in docs, we show users how to build template_extension outside of our build env: https://docs.openvinotoolkit.org/2020.4/openvino_docs_IE_DG_Extensibility_DG_Building.html.
In order to make this CMakeLists.txt work as a standalone, target_compile_options is required.

Reply:
In this case suggest to use:

```cmake
set(CMAKE_CXX_STANDARD 11)
```